### ISSUE TITLE
[$500] [HOLD for payment 2023-12-08] [HOLD for payment 2023-12-07] Mention - Suggested user @ with closed account remains visible when click on it

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#32214]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.5-3
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Navigate to a group chat or #Room that has several members (>6)
3. Enter "@" in the compose box
4. Select user that have closed their account

## Expected Result:
When user with close account is selected the suggestion list should close

## Actual Result:
Suggested user with closed account remains visible when click on it

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/484c4022-106e-4828-8b74-17e9798ac09d

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @sakluger</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sakluger</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sakluger</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c3279b7c2be27d87</li>
        <li>Upwork Job ID: 1755287424781955072</li>
        <li>Last Price Increase: 2024-02-07</li>
    </ul>
</details>

**Comment by situchan** [comment]
@joelbettner I tested and confirmed reverting https://github.com/Expensify/App/pull/31435 fixed this issue.
Test branch: https://github.com/Expensify/App/pull/32220

**Comment by joelbettner** [comment]
Thank you @situchan!

**Comment by situchan** [comment]
@ishpaul777 your solution will cause regression if search for `displayName email`

@joelbettner I have solution:
We can sync filter logic with display logic.
https://github.com/Expensify/App/blob/ef65a0a53d7dad70426e1e714fb4cedf8de5b15f/src/pages/home/report/ReportActionCompose/SuggestionMention.js#L161-L162

So display logic is if text = alternateText, don't display alternativeText
Same applies to filter logic:

If display name and formatted login is same, just filter with display name
```
                if (searchValue && !`${detail.displayName}`.toLowerCase().includes(searchValue.toLowerCase())) {
```
If different, use current logic

**Comment by joelbettner** [comment]
@allroundexperts if you wouldn't mind taking a look at the proposals that'd be great. I have some other time sensitive stuff I need to get worked on at the moment. Feel free to ping me when I should jump back in here.

**Comment by allroundexperts** [comment]
I think @situchan's [proposal](https://github.com/Expensify/App/issues/32214#issuecomment-1832700606) is better. @joelbettner Should I raise a PR or will @situchan do that?

**Comment by joelbettner** [comment]
@allroundexperts we can have @situchan raise the PR. Then you and I can review it.

**Comment by joelbettner** [comment]
@sakluger I'm not sure what needs to be done to close out this issue (I'm not able to make payments). Can you take a look?

**Comment by sakluger** [comment]
I'm still a bit confused about this one. @situchan said that https://github.com/Expensify/App/issues/32280 is a regression from https://github.com/Expensify/App/issues/30217, but I think that this issue also should have fixed it, right? I'm hesitant to pay this issue out quite yet while we're still discussing how to fix https://github.com/Expensify/App/issues/32280.

**Comment by sakluger** [comment]
@allroundexperts what are your thoughts on this one? Should this PR have fixed the issue we're seeing in https://github.com/Expensify/App/issues/32280?

**Comment by sakluger** [comment]
I think everyone has been offline for the holidays. I'll check in with @allroundexperts tomorrow on Slack to see how we should handle payment.

**Comment by sakluger** [comment]
Asked for help in Slack here: https://expensify.slack.com/archives/C01GTK53T8Q/p1704231902433439

**Comment by allroundexperts** [comment]
@sakluger I think that https://github.com/Expensify/App/issues/32280 should have been fixed in this PR. But we missed the case mentioned in https://github.com/Expensify/App/issues/32280 in the PR for this issue. I think the payment here should be halved.

**Comment by sakluger** [comment]
Got it, thanks! Based on the [Contributing guide](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#regressions), payments will be issued 7 days after all regressions are fixed (ie: deployed to production). We still don't have PR for that follow up issue, so we still can't pay this one out.

**Comment by sakluger** [comment]
Switching to Weekly while we are still working on fixing the regression.

**Comment by sakluger** [comment]
Still figuring out the regression.

**Comment by sakluger** [comment]
No update, I will prioritize this soon.

**Comment by sakluger** [comment]
There's still not agreement over whether https://github.com/Expensify/App/issues/32280 is a regression of this issue - given the uncertainty, I'm going to pay this one out in full.

**Comment by sakluger** [comment]
Summarizing payment on this issue:

Contributor: @situchan $500 ([Upwork offer](https://www.upwork.com/nx/wm/offer/100810202))
Contributor+: @sobitneupane $500, please request on Newdot

Upwork job is here: https://www.upwork.com/jobs/~01c3279b7c2be27d87

**Comment by sakluger** [comment]
Paid @situchan. @sobitneupane feel free to request payment. Thanks everyone for your patience on this one!

**Comment by sobitneupane** [comment]
@allroundexperts Did you work on this issue? I don't remember working on it.😀

**Comment by allroundexperts** [comment]
@sakluger I reviewed this PR instead of @sobitneupane. Also, please go through https://github.com/Expensify/App/issues/32214#issuecomment-1877261378 and let me know if you agree.

**Comment by allroundexperts** [comment]
@sakluger Bump for the updated payment summary!

**Comment by JmillsExpensify** [comment]
Re-opening for the updated payment summary.

**Comment by sakluger** [comment]
Sorry for the payment summary mixup, I've reposted the correct payment summary is below. @allroundexperts regarding the payment amount, there were reasonable arguments on both sides of whether this was a regression or not. Given the uncertainty, I decided that we would pay the full original amount.

**Summarizing payment on this issue:**

Contributor: @situchan $500 (paid via Upwork)
Contributor+: @allroundexperts $500, payable via Newdot
